src/string_id.h

```diff
@@ -3,6 +3,8 @@
 
 #include <string>
 #include <type_traits>
+#include <memory>
+#include "debug.h"
```

Comment:
Why are these includes needed here? I can't see anything in this files that uses their content.